toolchain/check/impl_lookup.cpp

```diff
@@ -451,65 +451,38 @@ auto LookupImplWitness(Context& context, SemIR::LocId loc_id,
   return context.inst_blocks().AddCanonical(result_witness_ids);
 }
 
-auto DeferredLookupSingleImplWitness(Context& context, SemIR::LocId loc_id,
-                                     SemIR::ImplSymbolicWitness deferred_query)
-    -> LookupSingleImplWitnessResult {
-  SemIR::ConstantId query_self_const_id =
-      context.constant_values().Get(deferred_query.query_self_inst_id);
-  SemIR::SpecificInterfaceId query_specific_interface_id =
-      deferred_query.query_specific_interface_id;
-
-  // NOTE: Do not retain this reference to the SpecificInterface obtained from a
-  // SpecificInterfaceId. Doing impl lookup does deduce which can do more impl
-  // lookups, and impl lookup can add a new SpecificInterface to the store which
-  // can reallocate and invalidate any references held in here.
-  auto query_specific_interface =
-      context.specific_interfaces().Get(query_specific_interface_id);
-
-  bool query_is_concrete = [&] {
-    if (!query_self_const_id.is_concrete()) {
+static auto QueryIsConcrete(Context& context, SemIR::ConstantId self_const_id,
```

Comment:
Maybe good to add a comment here (I'll pretty much always ask for function comments)? Here's how I've been reading this (and maybe I'm wrong):

```suggestion
// Returns true if the query will return a concrete constant.
static auto QueryIsConcrete(Context& context, SemIR::ConstantId self_const_id,
```

Reply:
Went with:
```
// Returns whether the query is concrete, it is false if the self type or
// interface specifics have a symbolic dependency.
```